parlai/core/torch_generator_agent.py

```diff
@@ -882,14 +882,24 @@ def eval_step(self, batch):
                 )
 
         preds = None
+        top_n_texts: List[List[Tuple[str, float]]] = []
         if self.skip_generation:
             warn_once("--skip-generation true produces limited metrics")
         else:
             maxlen = self.label_truncate or 256
-            beam_preds_scores, _ = self._generate(batch, self.beam_size, maxlen)
+            beam_preds_scores, beams = self._generate(batch, self.beam_size, maxlen)
             preds, scores = zip(*beam_preds_scores)
             self._add_generation_metrics(batch, preds)
 
+            for beam in beams:
+                top_n_texts.append([])
+                for tokens, score in beam.get_rescored_finished():
+                    try:
+                        top_n_texts[-1].append((self._v2t(tokens), score.item()))
+                    except KeyError:
+                        logging.error("Decoding error: %s", tokens)
```

Comment:
i know i was pulled in for naming but I am curious when this error arises?

Reply:
During beam search with high large size on model with low PPL that predicts not NLP, but structured messages sometimes things like `__FP16_PAD_0__` are produced that cause exception in v2t that I can't nicely avoid.